### PR TITLE
chore: require setting the identity - issuer and subject

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: hectorj2f <hectorf@vmware.com>

#### Summary

related to #107 

I am just checking what we want to enforce with this Draft pull request, some e2e tests need to be adapted to this new validation.

### CODE REVIEW

**Comment by hectorj2f** [comment]
I am working on adding warnings instead of errors following https://github.com/knative/pkg/pull/2498

File: pkg/apis/policy/v1beta1/clusterimagepolicy_validation.go [hunk 1]
```diff
@@ -226,7 +228,7 @@ func (p *Policy) Validate(ctx context.Context) *apis.FieldError {
 func (identity *Identity) Validate(ctx context.Context) *apis.FieldError {
 	var errs *apis.FieldError
 	if identity.Issuer == "" && identity.Subject == "" && identity.IssuerRegExp == "" && identity.SubjectRegExp == "" {
-		errs = errs.Also(apis.ErrMissingField("issuer", "subject", "issuerRegExp", "subjectRegExp"))
+		return apis.ErrMissingField("issuer", "subject", "issuerRegExp", "subjectRegExp")
```
Comment (hectorj2f) [inline pkg/apis/policy/v1beta1/clusterimagepolicy_validation.go:231]: i doesn't make sense to evaluate the fields if all are empty.

**Review by vaikas — APPROVED** [review]
Awesomesauce! Thanks for doing this! We should maybe add a section for these in our e2e tests as well. I added some code that you can create a CIP and add an expected error that gets validated. Here's where it's checked:
https://github.com/sigstore/policy-controller/blob/main/test/e2e_test_policy_crd.sh#L38

A sample entry for an error here:
https://github.com/sigstore/policy-controller/blob/main/test/testdata/policy-controller/invalid/keylessref-with-empty-authorities.yaml#L15

So, maybe we can add test/testdata/policy-controller/warn directory and add some CIPs there that produce warnings, but instead of ERROR: maybe have a section like WARN: and make sure the warning comes through.

WDYT? We could do that obvs in a followup.

**Comment by hectorj2f** [comment]
@vaikas it sounds like a plan to me 👍🏻

**Pull request merged**